packages/cli/src/metadataGeneration/parameterGenerator.ts

```diff
@@ -224,11 +224,7 @@ export class ParameterGenerator {
 
   private getFormFieldParameter(parameter: ts.ParameterDeclaration): Tsoa.Parameter {
     const parameterName = (parameter.name as ts.Identifier).text;
-    const type: Tsoa.Type = { dataType: 'string' };
-
-    if (!this.supportPathDataType(type)) {
-      throw new GenerateMetadataError(`Parameter '${parameterName}:${type.dataType}' can't be passed as form field parameter in '${this.method.toUpperCase()}'.`, parameter);
-    }
```

Comment:
I am hesitant to commit to this as is. While it enables a specific use case it's very likely to be a pitfall more often than not. We should at least warn that the user has to take additional steps here.

Reply:
I think we should use this instead of the error: https://github.com/lukeautry/tsoa/blob/master/packages/cli/src/metadataGeneration/exceptions.ts#L13

Reply:
Hello, I'm a bit lost about of that check: the `supportPathDataType` function accepts a lot for types that are not really managed by form fields no?

Regarding the check, is it possible to do it in a place where we can check if a JSON related type has been used without parsing the data?

Reply:
`supportPathDataType` basically rejects types that are serialized into objects.
Initially that was intended for path data parsing, but the assumption for form fields was similar: By default, form fields values will be `string`s, which can trivially be coerced into simple types, but not into arrays, or objects, or a mix of both.
Since I don't think middleware will try to parse form field values into objects, the check makes sense (with a warning), unless we commit to trying to coerce a form field's string value as a json object during validation.